tests/server/query_params/test_sort.py

```diff
@@ -98,3 +98,71 @@ def test_datetime_desc(get_good_response, structures):
         struct.get("attributes", {}).get("last_modified") for struct in response["data"]
     ]
     assert last_modified_list == expected_last_modified
+
+
+def test_unknown_field_errors(check_error_response, structures):
+    """ If any completely unknown field is provided, check 400: Bad Request is returned. """
+    limit = 5
+    request = f"/structures?sort=field_that_does_not_exist&page_limit={limit}"
+    check_error_response(
+        request,
+        expected_status=400,
+        expected_title="Bad Request",
+        expected_detail="Unable to sort on unknown field 'field_that_does_not_exist'",
+    )
+
+    request = f"/structures?sort=field_that_does_not_exist,-other_field_that_does_not_exist&page_limit={limit}"
+    check_error_response(
+        request,
+        expected_status=400,
+        expected_title="Bad Request",
+        expected_detail="Unable to sort on unknown fields 'field_that_does_not_exist', 'other_field_that_does_not_exist'",
+    )
+
+    request = f"/structures?sort=field_that_does_not_exist,nelements&page_limit={limit}"
+    check_error_response(
+        request,
+        expected_status=400,
+        expected_title="Bad Request",
+        expected_detail="Unable to sort on unknown field 'field_that_does_not_exist'",
+    )
+
+    # case 6: non-existent provider field
+    request = f"/structures?sort=_exmpl_provider_field_that_does_not_exist,nelements&page_limit={limit}"
+    check_error_response(
+        request,
+        expected_status=400,
+        expected_title="Bad Request",
+        expected_detail="Unable to sort on unknown field '_exmpl_provider_field_that_does_not_exist'",
+    )
+
+
+@pytest.mark.filterwarnings("ignore", category="FieldNotRecognised")
```

Comment:
Annoyingly, this doesn't seem to do anything...

Reply:
No, unfortunately it doesn't seem to respect custom categories. I would either use `capfd` or `capsys` and try and find it in `stderr`. Or `with pytest.warns(SomeCustomWarning):`?

Reply:
But if you want to use the decorator, your cold try and go for the message instead?

Reply:
> Or `with pytest.warns(SomeCustomWarning):`?

I'll try this, but I'm not sure how it interacts with the TestClient etc... would be cool if it works. Otherwise I'll try add a regexp for the message itself

Reply:
`pytest.warns` works beautifully, good shout